torch/csrc/autograd/functions/accumulate_grad.h

```diff
@@ -50,6 +50,26 @@ struct TORCH_API AccumulateGrad : public Node {
         // holding references to the Tensor and that is fine since these are not
         // exposed to the user.
         update_grad(new_grad.detach());
+      } else if (
+          !GradMode::is_enabled() && new_grad.is_sparse() &&
+          new_grad._indices().is_contiguous() &&
+          new_grad._values().is_contiguous() &&
+          new_grad._indices().use_count() <= num_expected_refs &&
```

Comment:
Is that correct? Shouldn't `num_expected_refs` be 1 here?

Reply:
These will have extra refs to the sparse Tensors. But the indices and values Tensors should only be referenced by the sparse Tensor.

Reply:
Yes, you are right. I'll fix this in a follow up PR since this one is already landing.

Reply:
how do you know indices and values only have 1 refcount?  What if you have a pre-existing reference to the indices tensor, from before you created the sparse tensor?

Reply:
ah ok, *checking* 1 should be okay.